src/playlist.js

```diff
@@ -232,14 +232,15 @@ export const sumDurations = function(playlist, startIndex, endIndex) {
   * @return {TimeRanges} the periods of time that are valid targets
   * for seeking
   */
-export const seekable = function(playlist, expired) {
+export const seekable = function(playlist) {
   let start;
   let end;
   let endSequence;
-
-  if (typeof expired !== 'number') {
-    expired = 0;
-  }
+  let expiredSyncPoint;
+  let firstSyncPoint;
+  let expired;
+  let expiredStart;
+  let segments;
```

Comment:
Wow, that is a lot of state for one function! 😮  I feel that at that point it might make more sense to break this into a few smaller functions.

Also the same comment as above about creating these inline.